lib/postgrex/notifications/connection.ex

```diff
@@ -0,0 +1,231 @@
+defmodule Postgrex.Notifications.Connection do
+  @moduledoc """
+  API for notifications (pub/sub) in Postgres.
+  """
+
+  use Connection
+
+  require Logger
+
+  alias Postgrex.Protocol
+  alias Postgrex.ConnectionUtils
+
+  @timeout 5000
+
+  defstruct protocol: nil, parameters: nil,
+            listeners: HashDict.new(), listener_channels: HashDict.new()
+
+  ## PUBLIC API ##
+
+  @doc """
+  Start the notification connection process and connect to postgres.
+
+  ## Options
+
+    * `:hostname` - Server hostname (default: PGHOST env variable, then localhost);
+    * `:port` - Server port (default: 5432);
+    * `:database` - Database (required);
+    * `:username` - Username (default: PGUSER env variable, then USER env var);
+    * `:password` - User password (default PGPASSWORD);
+    * `:parameters` - Keyword list of connection parameters;
+    * `:timeout` - Connect timeout in milliseconds (default: `#{@timeout}`);
+    * `:ssl` - Set to `true` if ssl should be used (default: `false`);
+    * `:ssl_opts` - A list of ssl options, see ssl docs;
+    * `:socket_options` - Options to be given to the underlying socket;
+    * `:sync_connect` - Block in `start_link/1` until connection is set up (default: `false`)
+    * `:extensions` - A list of `{module, opts}` pairs where `module` is
+      implementing the `Postgrex.Extension` behaviour and `opts` are the
+      extension options;
```

Comment:
We should refer to `Connection.start_link` instead of copying the options here, too easy to forget changing both places.